### PR TITLE
KIALI-2454 Add service validations information to the UI

### PR DESCRIPTION
**Pull request by burmanm**
** Describe the change **

Adds validation status for service validations in the service list. Waits for further information from kiali-design #118

** Issue reference **

KIALI-2454

** Screenshot **

In the kiali-design ticket.

### CODE REVIEW

**Comment by hhovsepy** [comment]
@burmanm  does it show the validation now, or still WIP?

**Comment by burmanm** [comment]
It shows it in the "service list", not in the details.

File: src/pages/ServiceList/ItemDescription.tsx [hunk 1]
```diff
@@ -5,20 +5,23 @@ import { ServiceHealth } from '../../types/Health';
...
+  validation: ObjectValidation;
 }
 
 export default class ItemDescription extends React.PureComponent<Props, State> {
   private promises = new PromisesRegistry();
 
   constructor(props: Props) {
     super(props);
-    this.state = { health: undefined };
+    this.state = { health: undefined, validation: {} as ObjectValidation };
```
Comment (xeviknal) [inline src/pages/ServiceList/ItemDescription.tsx]: Perhaps I am not fully following this. I see that the validation object is defined in the state interface and there isn't any instruction writting any value different thatn {}. Who is the one adding the actual Validations here?

File: src/pages/ServiceList/ItemDescription.tsx [hunk 2]
```diff
@@ -57,6 +60,14 @@ export default class ItemDescription extends React.PureComponent<Props, State> {
         <Col xs={12} sm={12} md={4} lg={4}>
           {!this.props.item.istioSidecar && <MissingSidecar />}
         </Col>
+        <Col xs={12} sm={12} md={4} lg={4}>
+          <strong>Config: </strong>{' '}
+          <ConfigIndicator
+            id={this.props.item.name + '-config-validation'}
+            validations={[this.state.validation]}
```
Comment (xeviknal) [inline src/pages/ServiceList/ItemDescription.tsx]: Won't we always have {} values? I guess that having the following would be enough to fix this:
```typescript
validations={[this.props.item.validation]}
```
And the state could be out of the state.
Does this make sense?

Comment (burmanm) [reply src/pages/ServiceList/ItemDescription.tsx]: Yeah, it was supposed to be that this.props.item.validation and no state at all, that was just the UI testing code.

**Comment by burmanm** [comment]
Rebased.

**Comment by hhovsepy** [comment]
Error config icon is in service list:
Screenshot from 2019-04-17 14-58-04

**Pull request merged**